extension/js/common/api/email-provider/gmail/google-auth.ts

```diff
@@ -121,7 +121,11 @@ export class GoogleAuth {
     }
     const authRequest: AuthReq = { acctEmail, scopes, csrfToken: `csrf-${Api.randomFortyHexChars()}` };
     const url = GoogleAuth.apiGoogleAuthCodeUrl(authRequest);
-    const oauthWin = await windowsCreate({ url, left: 100, top: 50, height: 800, width: 550, type: 'popup' });
+    const adaptiveHeight = Math.floor((window.innerHeight || window.screen.height) * 0.9);
+    const adaptiveWidth = Math.floor((window.innerWidth || window.screen.width) * 0.4);
```

Comment:
Thank you, @martgil.
I tested this in various multi-screen setups and it works ok, however, I think, we'd better calculate the adaptive width and height based on the screen boundaries (e.g. `const screenHeight = window.screen.height || window.innerHeight` etc.) as the popup window can be placed beyond the browser window (and it actually gets placed this way in certain conditions). This is how it looks now when there is a small browser window on a big screen.
![image](https://user-images.githubusercontent.com/23335944/183412513-50f54fba-a725-4536-8959-b42af8167c1a.png)
We can also increase the adaptiveWidth up to `* 0.9` if it is less than the comfortable width (550?)
We can also extract the non-standard properties without using `any` via `unknown` conversion and please add an explanation
```
// non-standard but supported by most of the browsers
const { availLeft, availTop } = (window.screen as unknown as { availLeft: number, availTop: number });
```

Reply:
Even better:
```
const { availLeft, availTop } = (window.screen as unknown as { availLeft?: number, availTop?: number });
```

Reply:
Thanks a lot for your help, sir! It now looks even better. I can now notice that prioritizing `window.innerHeight` creates a smaller OAuth window when the browser window is not in fullscreen mode (maximized). Code snippet for getting specific properties from `window.screen` + using unknown type is helpful -- I can use this later.